analyses/infercnv-consensus-cell-type/exploratory-notebooks/SCPCP000004/01_nb-consensus-cell-types.Rmd

```diff
@@ -0,0 +1,238 @@
+```
+
+We'll first exclude any libraries that have fewer than 1000 cells from consideration in the reference.
+
+```{r}
+discard_libraries <- celltype_df |>
+  dplyr::count(library_id) |>
+  dplyr::filter(n < 1000) |>
+  dplyr::pull(library_id)
+
+discard_libraries
+```
+
+```{r}
+celltype_df <- celltype_df |>
+  dplyr::filter(!(library_id %in% discard_libraries))
+```
+
+These cell groupings are present in the neuroblastoma samples we're considering:
+
+```{r}
+unique(celltype_df$reference_cell_group)
+```
+
+We should remove any cell groups which might correspond to tumor cells, including `stromal`, `neuronal`, and `stem cell`.
+The `NA` values here correspond to `Unknown` cells, which we'll also remove.
+This means we'll keep groups `endothelial`, `epithelial`, and `immune`.
+
+```{r}
+exclude_groups <- c("stromal", "neuronal", "stem cell")
+celltype_df <- celltype_df |>
+  dplyr::filter(
+    !(reference_cell_group %in% exclude_groups),
+    consensus_annotation != "Unknown"
+  )
+```
+
+
+
+Let's begin by counting how many cells in each cell group we have per library
+
+```{r}
+celltype_count_df <- celltype_df |>
+  dplyr::count(library_id, reference_cell_group)
+
+# format for viewing
+celltype_count_wide_df <- celltype_count_df |>
+  tidyr::pivot_wider(
+    names_from = reference_cell_group,
+    values_from = n,
+    values_fill = 0
+  ) |>
+  # add column for total normal cells
+  dplyr::mutate(
+    all_normal = rowSums(dplyr::across(where(is.numeric)))
+  ) |>
+  # arrange by total normal cells
+  dplyr::arrange(desc(all_normal))
+
+# this way we can print the full table at once; RMarkdown will only show 10 rows at once
+knitr::kable(celltype_count_wide_df)
+```
+
+Virtually all of the normal cell types are `endothelial` and `immune`, with very few `adipocyte` and literally only one `epithelial` cell.
+
+We'll look at this data in barplot form to see, per-library, the relative proportions of each normal cell type group out of all normal cells in the library.
+
+
+```{r}
+celltype_prop_df <- celltype_count_wide_df |>
+  # long again, but keep all_normal column for calculating proportion
+  tidyr::pivot_longer(
+    -c(library_id, all_normal),
+    names_to = "reference_cell_group",
+    values_to = "n"
+  ) |>
+  dplyr::mutate(cell_group_prop = n / all_normal)
+
+ggplot(celltype_prop_df) +
+  aes(x = library_id, y = cell_group_prop, fill = reference_cell_group) +
+  geom_col() +
+  labs(y = "Proportion of normal cells in group") +
+  theme(
+    legend.position = "bottom",
+    axis.text.x = element_text(angle = 30, hjust = 1, size = 7)
+  )
+```
+
+Based on these numbers, we can consider three normal reference options:
+
+* All normal cells
+* ALl immune cells
+* All endothelial cells
+
+How many cells would be in each of those pooled references?
+
+```{r}
+# update the count data frame without categories we don't need
+celltype_count_df <- celltype_count_wide_df |>
+  dplyr::select(-adipocyte, -epithelial) |>
```

Comment:
To be clear, these are still included in the `all_normal` count but just not being printed out as rows in the below table?

Reply:
Yes - they are still included in `all_normal`! But we don't need them as standalone categories. I'll update the comment to make this clear.